internal/notifier/adapters/scheduler.go

```diff
@@ -1,34 +1,56 @@
 package adapters
 
 import (
-	"go_service/internal/infrastructure/database/models"
-	"gorm.io/gorm"
+	"encoding/json"
+	"os"
 	"time"
 )
 
-type ScheduleDBAdapter struct {
-	db *gorm.DB
+type TimeData struct {
+	LastSentTime string `json:"last_sent_time"`
 }
 
-func NewScheduleDBAdapter(db *gorm.DB) *ScheduleDBAdapter {
-	return &ScheduleDBAdapter{db: db}
+type ScheduleAdapter struct {
+	filename string
 }
 
-func (sa ScheduleDBAdapter) GetLastTime() *time.Time {
-	var lastTime models.ScheduleTime
-	result := sa.db.Last(&lastTime)
-	if result.Error != nil {
+func NewScheduleAdapter() *ScheduleAdapter {
+	return &ScheduleAdapter{filename: "conf/email_sent_time.json"}
```

Comment:
Where is a file?

Reply:
This is an automatically generated file with persistent information about the last email sent so that the notifier knows where it sent the last emails after restarting.